### DISCUSSION TITLE
Build on Windows

### DISCUSSION THREAD

**Discussion by C-BJ** [hylo-lang/hylo#738] [Q&A] [answered]
I tried to build Val on Windows, I just downloaded Swift but everything seems to work fine, it's not as complicated as [#45](https://github.com/val-lang/val/issues/45) says(Perhaps the documentation needs updating? Or am I already setting something up that I don't know about?). 

The ast and cpp files are generated normally. The only problem is that the following problems occur when binary files are generated.
![image](https://user-images.githubusercontent.com/62678643/214496007-a49ef64c-3e86-4816-9481-5f7a6b0f90fa.png)

**Comment by kyouko-taiga** [comment, marked as answer]
Hello Cai!

Thanks a lot for this report.I will update the README accordingly.

Just a hunch, but I suspect the current code for locating `clang` is broken on Windows (see [Sources/CLI/main.swift:342](https://github.com/val-lang/val/blob/2f107ae0e8e711f7ff91169fd57d01c0a43e9e06/Sources/CLI/main.swift#L342)). May I ask you to open an issue on the compiler repository about this problem?

**Comment by C-BJ** [comment]
One question, does Val require Clang to use `libc++`?

**Comment by kyouko-taiga** [comment]
I think so, since we're currently transpiring Val to C++

**Comment by C-BJ** [comment]
Can I use other libraries, such as Microsoft's or `libstdc++`? @kyouko-taiga

**Reply by kyouko-taiga** [reply]
As of now, we're not linking anything fancy when compiling generated C++. In fact, I think any C++ compiler should be able to ingest the code we transpile, not only clang.

We should eventually implement support to customize the toolchain, so that users can use their preferred compiler. We chose clang initially out of convenience, as most of the development is done on macOS.

**Reply by C-BJ** [reply]
OK, But I may not be very clear saying this, I want to say is there any problem with Val using Clang with other standard libraries(isn't libc++), but there seems to be no problem like you say. thanks.

**Comment by C-BJ** [comment]
Is a lot of `ld-link: warning` normal when compiling? @kyouko-taiga

**Reply by kyouko-taiga** [reply]
I assume you're asking about building on Windows? I'm afraid I don't know much about that, as I didn't do anything regarding that OS myself. The current state of our Windows support is a kind contribution of @compnerd. Maybe they can shed some light?

**Reply by compnerd** [reply]
Unfortunately yes, that is expected.  That is a limitation of SPM, and I do hope to look into that soon.

**Reply by C-BJ** [reply]
https://github.com/apple/swift/issues/56030
